### DISCUSSION TITLE
Feature Request: Separate Configurations for Quick Terminal

### DISCUSSION THREAD

**Discussion by dljsjr** [ghostty-org/ghostty#3524] [Feature Requests, Ideas]
First: Ghostty is incredible. I'm an immediate convert; thank you so much for sharing this labor of love!

Second: Related to the comments on https://github.com/ghostty-org/ghostty/issues/2331, I wanted to make an official request for being able to have separate style options for the Quick Terminal.

My personal use-cases specifically are the default window height and background opacity. Quake-style terminals are such an integral part of my workflow that I even ended up [patching Kitty](https://github.com/kovidgoyal/kitty/issues/7145) to add the feature. Though I haven't gotten around to upstreaming with Kovid's requested changes 😅. Having the drop-down terminal be transparent is pretty crucial for me.

Opening this issue to see if we can get additional input to achieve the "critical mass" of requests mentioned in #2331's comments/discussion.

**Comment by Integralist** [comment]
I accidentally made a duplicate https://github.com/ghostty-org/ghostty/discussions/3697 and then closed it when I saw this.

EDIT: There's also https://github.com/ghostty-org/ghostty/issues/2384

**Comment by keldin-coding** [comment]
Oh i would love configurable opacity. I've been enjoying the light usage of Ghostty so far, but I really do miss my dropdown window being somewhat transparent. I know https://github.com/ghostty-org/ghostty/issues/2384 is approaching quick terminal size (yay!), but i really do like it being transparent. Reminds me that it's more for quick, ephemeral work rather than my main work terminal. While I'd also love tabs in the quick terminal, I'm okay with that being off the table as that seems more intended vs "let's wait and see if the config complexity is worth it."

**Comment by joshleaves** [comment]
Seconding this, but now I see the weight of the task.

I've started looking into the code and wrote a patch that made `QuickTerminalController.swift` pull from another config variable, but in the end, the value is set up in `Metal.zig` and `OpenGL.zig` and always pulled from `config.@"background-opacity"`. To differentiate a regular terminal and a quick terminal would require to bubble all the way from (both) controllers, through the Surface, down to the Renderer classes.

If anyone feels like looking at the problem, feel free to build from my patch, I'd hate to see my first code of the year go to waste :p

[quick_terminal_background_opacity.patch](https://github.com/user-attachments/files/18286692/quick_terminal_background_opacity.patch)

**Comment by joshleaves** [comment]
For those who'd like to start their Quick Terminal in another folder, or run another shell setup configuration, here is an option: abuse the terminal size property. Using `tputs lines` will output the number of lines in the terminal, so you can use that to differentiate the QT and your regular windows on opening.

So, by adding the following to your shell config (in my case, it's [fish](https://github.com/fish-shell/fish-shell), ask ChatGPT for conversion), you will be able to branch out:

```shell
if test $(tput lines) -lt 25
  source visor-only-config.sh
end
```

Note: I put the size of mine at 300px, that's 20 lines. Should work for most cases, unless you use a very big QT.
